chain/x/zoracle/handler.go

```diff
@@ -108,7 +133,20 @@ func handleEndBlock(ctx sdk.Context, keeper Keeper) sdk.Result {
 			continue
 		}
 
-		result, _, errOwasm := owasm.Execute(&env, script.Code, "execute", request.Calldata, 100000)
+		result, gasUsed, errOwasm := owasm.Execute(
+			&env, script.Code, "execute", request.Calldata, request.ExecuteGas,
+		)
+
+		if gasUsed > request.ExecuteGas {
+			gasUsed = request.ExecuteGas
+		}
+
+		gasConsumed, overflow = addUint64Overflow(gasConsumed, gasUsed)
+		// Don't expect to happen
```

Comment:
Let's make the comment a bit clearer. 

// Must never overflow because we already checked for overflow above with gasConsumed + request.ExecuteGas (which is >= gasUsed).